tests/test-kicker.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+# This test exercises the kicker script functionality by reading in sample data
+# containing 15 (defanged) ch202 entries and writing them each to a separate file
+# in a temporary directory.  For each file, we echo its name into test-kicker-json.out
+# (followed by a newline), and remove it.  Once all files have been kicked
+# (resulting in 15 lines in test-kicker-json.out) the total line count must equal 15.
+
+script_dir=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+nmsgtool_test=$script_dir/../src/nmsgtool
+infile=$script_dir/private-tests/202_sample.json
+outdir=/tmp/test-kicker/
+outfile=$outdir/test-kicker-json.out
+kicker="echo && rm"
+retval=0
+
+check() {
+	if [ $? = "0" ]; then
+		echo "PASS: $*"
+	else
+		echo "FAIL: $*"
+		retval=1
+	fi
+}
+
+echo "Testing kicker: "
+
+mkdir $outdir
+
+# Read input data with our kicked nmsgtools.
+$nmsgtool_test -ddddd -j $infile -c 1 -k "$kicker" > $outfile
+file_count=$(wc -l $outfile | awk '{print $1}')
+entry_comparison=0; [ "$file_count" == "15" ] && entry_comparison=1
```

Comment:
This notation is a bit cryptic... I think simply `[ "$file_count" == "15" ]` should suffice, no?

Reply:
This is indeed using the kicker functionality under the hood but is it really testing it? The input file has 15 lines; the output has 15 lines. For all we know, it could simply be regurgitating the input without applying any transformation. I think a more reasonable test might be running this in a fresh test directory, making sure that directory is first empty, and then checking to see that 15 files have been created which match a glob of say... the current year/month/day/etc. (and of course, deleting them all afterwards)